src/v2/generated.rs

```diff
@@ -2176,3 +2202,18 @@ impl TryFrom<TokenomicsInfo> for super::types::RewardsOverview {
         }
     }
 }
+
+impl TryFrom<Branch> for super::types::queries::Branch {
+    type Error = tonic::Status;
+
+    fn try_from(value: Branch) -> Result<Self, Self::Error> {
+        Ok(Self {
+            block_hash: value.block_hash.require()?.try_into()?,
+            children:   value
+                .children
+                .into_iter()
+                .map(|c| c.try_into())
+                .collect::<Result<_, _>>()?,
```

Comment:
I don't think this is a good instance. It is recursive, with no constraint on the depth of recursion. Please revise.

Reply:
I have rewritten the try_from to avoid recursion and added a unit test